### PR TITLE
Add config structure

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -1 +1,22 @@
...
+	GuidRangeEnd   string `json:"guid_range_end"`
+}
+
+type SubnetManagerPluginConfig struct {
+	Name        string `json:"subnet_manager_name"`
+	Username    string `json:"subnet_manager_username"`
+	Password    string `json:"subnet_manager_password"`
+	Address     string `json:"subnet_manager_address"`
+	Port        int    `json:"subnet_manager_port"`
+	HttpSchema  string `json:"subnet_manager_httpSchema"`
```
Comment (moshe010) [inline pkg/config/config.go]: what is define here? can we add comment?

File: pkg/config/config.go [hunk 2]
```diff
@@ -1 +1,22 @@
...
+	Certificate string `json:"subnet_manager_certificate"`
```
Comment (moshe010) [inline pkg/config/config.go]: what we expect here? can we had comment?

File: pkg/config/config.go [hunk 3]
```diff
@@ -1 +1,22 @@
...
+	SubnetManager  SubnetManagerPluginConfig
+}
+
+type GuidPoolConfig struct {
+	GuidRangeStart string `json:"guid_range_start"`
+	GuidRangeEnd   string `json:"guid_range_end"`
+}
+
+type SubnetManagerPluginConfig struct {
+	Name        string `json:"subnet_manager_name"`
```
Comment (moshe010) [inline pkg/config/config.go]: it better to hold each plugin config in it own struct.
So you will have UFM config for now in the future there can be more.
so 
type SubnetManagerPluginConfig struct {
UFMConfig
}
in the future :
type SubnetManagerPluginConfig struct {
UFMConfig
OtherPlugin
}

Comment (Mmduh-483) [reply pkg/config/config.go]: > ixed

File: pkg/config/config.go (hunk 2 above)
Comment (Mmduh-483) [reply pkg/config/config.go]: added

File: pkg/config/config.go (hunk 1 above)
Comment (Mmduh-483) [reply pkg/config/config.go]: added

File: pkg/config/config.go [hunk 4]
```diff
@@ -1 +1,26 @@
 package config
+
+type DaemonConfig struct {
+	Plugin         int `json:"plugin"`          // Subnet manager plugin name
```
Comment (moshe010) [inline pkg/config/config.go]: Plugin is int?

Comment (Mmduh-483) [reply pkg/config/config.go]: fixed

**Pull request merged**